framework/include/cppmicroservices/detail/ServiceTracker.hpp

```diff
@@ -139,8 +139,16 @@ void ServiceTracker<S, T>::Open()
 template<class S, class T>
 void ServiceTracker<S, T>::Close()
 {
+  // Copy required to prevent data race in call to context.RemoveListener()
+  ListenerToken tokenCpy;
+  {
+    auto l = d->Lock();
+    US_UNUSED(l);
+    std::swap(d->listenerToken, tokenCpy);
```

Comment:
The comment and this line of code don't match up. std::swap does exactly that, swaps the state of the two objects, effectively doing this:

    auto tmp = a;
    a = b;
    b = tmp;

So the state of d->listenerToken above is reset to the default state for the ListenerToken class.

Reply:
good catch. yes, the comment is not telling the truth